### DISCUSSION TITLE
Two practical design innovations to enhance user-experience and productivity

### DISCUSSION THREAD

**Discussion by b9Joker108** [community/community#110250] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

Dear GitHub and to whomever else it may concern 

I have a practical design enhancement and user-based innovation suggestion, that significantly impacts upon practical usefulness and usability, in document preparation on GitHub. I know this isn't the primary function of GitHub, but it holds much import for the medium and forum, as documentation is key to community and community capacity building, innovation, and global discourse in general. 

I am writing a document in GitHub Flavoured Markdown, with associated code, in the GitHub repository environment, that has become quite large, namely:

https://github.com/b9Joker108/a.fable.of.the.scriptkiddie.scriptoria/blob/main/A_Fable_of_the_Scriptkiddie_Scriptoria_(GitHub_test).md

It was a design choice, at this initial stage, to use a singular page design in Markdown, rather than a multi-node reticulum, forded through a multi-paged wiki, or a traditional HTML archive, connected by metatext/hyperlinks/hypertext. (Well, in truth, it is less a design choice, and more the stuff of happenstance, and that is just the way I draft best. The work is still very much a draft.)

I just noticed the chains icons, at the left of the various Markdown heading levels. I have been editing this document for some time. Well, in truth quite briefly, but only just noticed them and explored their functionality. Now, I do note and know, that I am blinkered when I write, as I am an impassioned writer and researcher. That said, I feel it unlikely that I haven't noticed these for the quite a few weeks or the very few months I have had an active GitHub account, and repositories: quite unlikely, though possible. If indeed, they are a newly implemented design enhancement, or otherwise, I humbly petition for two more. 

Now in the chain icons, that provide various functionalities, such as an intra-linkage specificity, for example:

https://github.com/b9Joker108/a.fable.of.the.scriptkiddie.scriptoria/blob/main/A_Fable_of_the_Scriptkiddie_Scriptoria_(GitHub_test).md#fizsh-a-zsh-shell-extension-subsuming-fish-like-features-and-experience

May we usefully piggyback in the chain icon dropdown options, the new functionality to jump to editing at that particular section at-once or with immediacy, rather than selecting editing mode globally, and then trawling or scrolling through the document for the relevant section to be edited. As the document has become large, this is now cumbersome and problematic, and an inexact science, as well as manifestly inefficient. I take inspiration from the Wikimedia Projects section editing functionality. I feel this would usefully enhance the GitHub Flavoured Markdown document writing process, experience, and subsequent efficiency. I do note that there is the wiki functionality as an option in repositories, and that, though I haven't done so, I intuit that migration from GitHub Flavoured Markdown to the GitHub Wiki forum, may usefully provide both the functions that I entreat. Though, I haven't yet explored them. 

In addition, may we please have a document or page search field, to find items, such as words or phrases, for example, within GitHub Flavoured Markdown and associated document file extensions, and to cycle through such items? 

These two design implementations would greatly enhance usability and functionality. I humbly petition for their inclusion. 

Thank you so much for the GitHub forum and medium. 

Thanking you in anticipation 
Beauford 

P. S. 

I just copied and paste this Markdown document intra-linkage:

https://github.com/b9Joker108/a.fable.of.the.scriptkiddie.scriptoria/blob/main/A Fable of the Scriptkiddie Scriptoria (GitHub test).md#fizsh-a-zsh-shell-extension-subsuming-fish-like-features-and-experience

And note, that though when I place it in the URL of an Internet browser it is directly discoverable, it functions as a broken link when pasting wholesale in this document, for your knowledge. I note, that it is possibly due to my ignorance that it is probably not working. I tried to repair the above linkages, and placed underscores where there are spaces, and though the linkage is now visually unbroken, it is still practically.

**Comment by MasterInQuestion** [comment]
Do proper URI encoding.
    For more info: https://pastebin.com/raw/YTi5UzLq
    [ Upward: [https://forum.vivaldi.net/topic/77666](https://forum.vivaldi.net/topic/77666) ]

    https://github.com/b9Joker108/a.fable.of.the.scriptkiddie.scriptoria/raw/main/A%20Fable%20of%20the%20Scriptkiddie%20Scriptoria%20(GitHub%20test).md
    The ~ 2 MiB text file is beyond reason, sincerely.

    And avoid using scheme that relied on syntax highlighting / post-rendering: poor general interoperability.

    Might be of use: [https://github<i>.</i>com/microsoft/vscode/issues/206902#issue-2169741980](https://github.com/microsoft/vscode/issues/206902#issue-2169741980)
    (exported outline of document)

    ----

    Summary:
    |1| "Search in file": Browsers should have Search in Page mostly.
    |2| "Section Edit for Markdown": Probably not much useful. (many GitHub operations are file-based)

**Reply by b9Joker108** [reply]
@MasterInQuestion I don't know how I have put your nose out of joint, or what your problem is with me or my project, but your manner is abrasive, verging on rude, if not outright rude. You, are of course entitled to your opinion. But, I find your understanding and view of the nature of the issues and contentions I raise, quite narrow and ill-suited, and most decidedly, and manifestly, unhelpful! If you are trying to be helpful, you are failing. I ask you to think outside your purview, personal scope and experience, and think of the usecase of others writ large, when responding to GitHub community matters. This would make your responses more useful.

Why are you posting my project, without my assent or consent, in a forum about VSCode on Microsoft? I find this perplexing, when my project is ONLY for Termux on Android, and for Neovim, specifically! I don't understand the nature of the issues you raise, such as: URI encoding, a scheme that uses syntax highlighting when that is determined by the specified language automatically within a codeblock with GitHub Flavoured Markdown. 

In addition, I meant the edit section functionality specifically for editing a document, and just used the wiki example, to provide a model for implementation. Moreover, the search function and being able to cycle through, would be very useful, many browsers do not have that function, let alone the GitHub Android app. Believe me, I know from personal experience. In my project, having the document only in a "simple text file", as you mentioned in my other post, is clearly not in keeping with my project outline, aims and project auspice. So, why you championed and ventured it, runs totally counter to the project design, which is modelled on a number of FOSS projects, which I value and hold in esteem. Also, GitHub already has the page search code solution, it is a functionality that is automatically enabled in webpages rendered with the GitHub workflow, of repository website's hosted on GitHub. So, the code algorithm would most likely be easily transferable. If I was to publish the website of my GitHub repository project in question, consisting principally of the page you say is "beyond reason", that would automatically have a GitHub designed and implemented, search field.

Another reason, why I have favoured leaving my document in one page for production, apart from all the other reasons outlined in my other post, is for generating both the TOC and index, with associated metatext/hyperlinks/hypertext. I am just now, commencing finding a solution for this.

**Reply by MasterInQuestion** [reply]
My apologies if it upsets people, but the truth hurts.
    I merely target the actual problems, never personal.
    And I'm not in the habit of wrapping my texts with Hello messages. (significant overhead)

    None of my design decisions is trivially chosen: all have undergone extensive analysis, and mostly verified against significant amount of data.

    Yourself made the project public, and linked it publicly.
    I quoted it merely as a sample that would cause the failure.
    (also exported your document outline in belief that would be helpful)

    URI encoding is specifically referring to the link you originally posted: broken unintended, as yourself noticed.
    You should perform proper URI encoding in order to not have which.

    "document", what document?
    Without considering the exact details, much would be meaningless.

    "many browsers do not have that function" (Search in Page)
<^>    Many of the minority browsers..?

    In addition, I never expressed any strong objection to the implementation of related features:
    Merely summarized them quickly with relevant hints.

    "beyond reason" merely reflects to the file size, not its content.
    Text files of such size typically cause serious interoperability issues.
    .
    Probably you have done some decent writing.
    But I haven't got the time to check.

    ----

    Given some quick check to its content (mostly outline-based), it appears:
    Bunch of separate titles gathered in poorly arranged hierarchy.
    .
    Reorganization is required, shall you consider accessibility and portability.

    Some caveats during the check:
    |&#x2A;| Use of ["m.wikipedia.org"](https://en.m.wikipedia.org/wiki/Main_Page) alike links (that coerces mobile page).
    |&#x2A;| Use of non-standard "#:~:text" ([Text Fragments](https://developer.mozilla.org/en-US/docs/Web/Text_fragments)) anchor extension. (lacking web compatibility and easily broken; and garbles the URL, mostly)
    |&#x2A;| Suboptimal use of Google Search URL: unnecessary parameters (mostly everything besides "q"), and significant ones ("hl", "gl") not set. [1]
    |&#x2A;| Similar suboptimal use of YouTube links. [ Canonical form: [ [https://www.youtube.com/watch?v=_P8PLHvZygo](https://www.youtube.com/watch?v=_P8PLHvZygo) ] alike. ]

    Also: Less verbosity = More readability
    In particular notable for titles, filenames alike.
    .
    Suggested new name: Scriptoria
    (rather than the "A Fable of the ...")

    Also poor lead: 2 one-liner like big text-blocks hardly summarize anything. (verbosity and off-topic)
    And your coined term "computational shell" doesn't make much sense. ("computer shell", or just "shell")
    Probably you also have misunderstanding on the concept: "shell". (rough parse of the lead indicated)
    .
    First 200 lines of the document hardly provide any information of interest. (bunch off-topic and greatly lost)
    What's the real topic of the subject? Shouldn't which be a technical guide..?
<img src="https://github.com/MasterInQuestion/attach/raw/main/community/community/110250/Comma-Splice.20.re.webp" alt="    [ &quot;Comma-Splice.webp&quot; ]" />

    And avoid misleading description like "by The Google Bard ... (a.k.a. b9Joker108 at GitHub)":
    Such would only render the texts discreditable.

[ [1]
    See also: [https://bugzilla.mozilla.org/show_bug.cgi?id=1813236#c1](https://bugzilla.mozilla.org/show_bug.cgi?id=1813236#c1)
    But I suspect linking regular words to Google Search as dictionary use is justifiable.
    .
    Such is more like, "over-hyped hyperlinking":
    [ https://github.com/orgs/community/discussions/68734#discussioncomment-7347910](https://github.com/orgs/community/discussions/68734#discussioncomment-7347910)

    Moreover, for this usage: the term should be &#x60;define:"how to"&#x60; alike; preferably with "expnd=1".
    E.g. https://www.google.com/search?hl=en&gl=ca&num=1&expnd=1&q=define:whatever ]

**Comment by b9Joker108** [comment]
I just recategorised this thread, from discussion to general.
